File: app/src/main/java/im/bernier/petfinder/BaseApplication.java
package im.bernier.petfinder;

import android.app.Application;

import com.google.firebase.analytics.FirebaseAnalytics;

import timber.log.Timber;

/**
 * Created by Michael on 2016-07-09.
 */

public class BaseApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        if (BuildConfig.DEBUG) {
            Timber.plant(new Timber.DebugTree());
        }
        Analytics.getInstance().init(FirebaseAnalytics.getInstance(this));
    }
}

File: app/src/main/java/im/bernier/petfinder/BaseApplication.kt
/*
 *   This Source Code Form is subject to the terms of the Mozilla Public
 *   License, v. 2.0. If a copy of the MPL was not distributed with this
 *   file, You can obtain one at http://mozilla.org/MPL/2.0/.
 *
 * If it is not possible or desirable to put the notice in a particular
 * file, then You may include the notice in a location (such as a LICENSE
 * file in a relevant directory) where a recipient would be likely to look
 * for such a notice.
 *
 * You may add additional accurate notices of copyright ownership.
 */

package im.bernier.petfinder

import android.app.Application

import com.google.firebase.analytics.FirebaseAnalytics

import timber.log.Timber

/**
 * Created by Michael on 2016-07-09.
 */

class BaseApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        if (BuildConfig.DEBUG) {
            Timber.plant(Timber.DebugTree())
        }
        Analytics.instance.init(FirebaseAnalytics.getInstance(this))
    }
}
